internal/testrunner/runners/pipeline/runner.go

```diff
@@ -108,6 +108,18 @@ func (r *runner) run() ([]testrunner.TestResult, error) {
 		expectedDataset = pkgManifest.Name + "." + r.options.TestFolder.DataStream
 	}
 
+	// when reroute processors are used, expectedDataset should be set to "" to avoid checking the dataset fields
+	for _, pipeline := range r.pipelines {
+		if processors, err := pipeline.Processors(); err == nil {
+			for _, p := range processors {
+				if p.Type == "reroute" {
+					expectedDataset = ""
```

Comment:
Should the expected dataset be one of the indicated in the reroute processor?

Reply:
Yes! Just changed the code to get expectedDataset from the reroute processors. Thanks!